cloudinit/net/__init__.py

```diff
@@ -928,6 +928,12 @@ def get_interfaces_by_mac(blacklist_drivers=None) -> dict:
         )
 
 
+def get_interface_from_mac(mac: str):
```

Comment:
No objections to adding `return None` for clarity, +1 to that.

> raw comparison -> regex: Other clients of this function won't have to deal with mac.lower().

I'm not sure I understand the argument here. Is this to allow potential future callers of this function to regex search for macs?

Reply:
At the moment, if the caller of the function does not convert mac to lower-case, this function is not going to work correctly.

We could document this fact, or perform a mac.lower() inside this function, in order to be defensive.

I am suggesting to use the regex to be defensive, avoid writing extra documentation and because I think it is more concise than a `mac.lower() == interface_mac.lower()`.

But it is good as it is, as the only caller of the function convert mac to lower case. Maybe I am overthinking here, if so, please ignore this.

Reply:
I actually prefer `mac.lower() == interface_mac.lower()` here. There's no actual regex pattern being searched here, so I'm not sure it makes sense to introduce regex matching. There's also no ambiguity. I'm guessing we're safe from interface names not containing any characters that could be interpreted as a regex pattern, but if we use `.lower()` we know for sure. 

Either solution is fine though.